### PR TITLE
CodeMirror update

### PR DESCRIPTION
**Pull request by zaggino**
PR for https://github.com/adobe/brackets/issues/12937

### CODE REVIEW

**Comment by ficristo** [comment]
Why not use requirejs to create some logical mapping between the node_modules and thirdparty folders?
I think @petetnt has done that in https://github.com/adobe/brackets/issues/12006
If we go on this route we probably need to do something for distribution. (I don't remember exactly but I think we don't copy the node_modules over there)

**Comment by zaggino** [comment]
Not sure how to do any logical mapping here, but happy to learn. We could use a separate package.json file inside the src folder for production dependencies. That's how brackets-electron does it and that's how electron authors recommend electron projects to be done.

**Comment by ficristo** [comment]
The other idea I had in mind was exactly putting a package.json inside the src folder.
Both cases seems better to me.
For the logical mapping @petetnt know more.

**Comment by petetnt** [comment]
One can use relative mapping with `../node_modules/...` for the `requirejs` config so you can use deps from the root folder: https://github.com/petetnt/brackets/blob/petetnt/poc-thirdparty-with-npm/src/main.js#L68. Not sure if that works in browser though.

File: Gruntfile.js [hunk 1]
```diff
@@ -323,7 +323,7 @@ module.exports = function (grunt) {
     });
 
     // task: install
-    grunt.registerTask('install', ['write-config', 'less', 'npm-install-extensions']);
+    grunt.registerTask('install', ['write-config', 'less', 'copy-browser-dependencies', 'npm-install-extensions']);
```
Comment (le717) [inline Gruntfile.js]: Fast note: the script is referenced here as `copy-browser-dependencies` but is actually named `move-browser-dependencies`.

**Comment by zaggino** [comment]
Closing in favor of https://github.com/adobe/brackets/pull/12972

**Pull request closed without merging**